### ISSUE TITLE
Optimize Object Pose with a PhongShader instead of SilhouetteShader

### ISSUE THREAD

**Issue by filipetrocadoferreira** [facebookresearch/pytorch3d#30]
I'm trying to optimize a mesh pose using a target image and the rendered. Because I'm working in synthetic domain I thought that I could use PhongShader instead of Silouette proxy.

However, right in first iteration weights explode. Is there a simple justification for this? Or maybe I'm doing something wrong?

**Comment by jcjohnson** [comment]
Can you post a bit of code that reproduces the problem? That will make it easier for us to help.

**Comment by nikhilaravi** [comment]
@filipetrocadoferreira I can take a look at this and get back to you. You may need to increase the `faces_per_pixel` for the PhongShader.

**Comment by filipetrocadoferreira** [comment]
I was having a hard time spliting the code to show here
So, I tried camera pose optimization tutorial and changed the SilhouetteShader with a Phong, and it worked.

In my problem, I estimate Rotation (R) from a neural network but after the first backpropagation, all values go to nan, but the loss is finite.

**Comment by nikhilaravi** [comment]
@filipetrocadoferreira ok great!!
